lib/modules/manager/pub/artifacts.ts

```diff
@@ -26,32 +29,45 @@ export async function updateArtifacts({
   newPackageFileContent,
   config,
 }: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
-  if (
-    !(
-      config.updateType === 'lockFileMaintenance' ||
-      config.updateType === 'lockfileUpdate'
-    ) &&
-    updatedDeps.length < 1
-  ) {
+  logger.debug(`pub.updateArtifacts(${packageFileName})`);
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+  if (!is.nonEmptyArray(updatedDeps) && !isLockFileMaintenance) {
```

Comment:
Avoids a double negative:
```suggestion
  if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
```